services/galexie: Update changelog for v23 release (#5787)
stellar/go: services/galexie/UPGRADE.md

# Upgrade Guide

## v23.0.0

> This release includes a breaking change to the datastore.

Reviewer: ```suggestion
This release includes a breaking change to how ledger files are stored in the data lake.
```

#### 1. The Change
   The ledger file extension has been updated from `.zstd` to the standard `.zst` for Zstandard compression.

#### 2. Datastore Update Options
   You have two options to update your datastore. Both will require you to start fresh and repopulate your entire data lake, as old files are no longer compatible with this version of Galexie. ⚠️ Galexie will refuse to upload files to any bucket that still contains `.zstd` files.

> **Option 1**: Use a New Datastore Location (Recommended)\

Reviewer: ```suggestion
**Option 1**: Use a New `destination_bucket_path` (Recommended)\
```

    This is the safest method to ensure no issues with the old files. Update the destination_bucket_path in your configuration file to a new, empty bucket, then restart Galexie.

**Option 2**: Clean Your Existing Datastore \
   If you want to use your current datastore, you must manually delete all files with the .zstd extension. This will permanently remove your existing data history. Then, restart Galexie.

#### 3. Post-Upgrade Verification

>    After restarting, confirm the following:

Reviewer: ```suggestion
   After before upgrading Galexie, confirm the following:
```

>    After restarting, confirm the following:

Author: ```suggestion
   After upgrading Galexie, confirm the following:
```

 - New ledger files are being written with the `.zst` extension.

>  - A `.config.json` manifest file has been created in your datastore location (this happens automatically).

Reviewer: ```suggestion
 - A `.config.json` manifest file has been created in `destination_bucket_path` (this happens automatically).
```
